src/probnum/randprocs/markov/_markov.py

```diff
@@ -107,27 +101,67 @@ def _sample_at_input(
             ]
         )
 
-    def push_forward(
+
+class MarkovProcess(_MarkovBase):
+    r"""Random processes with the Markov property.
+
+    A Markov process is a random process with the additional property that
+    conditioned on the present state of the system its future and past states are
+    independent. This is known as the Markov property or as the process being
+    memoryless. A Markov process can be fully defined via an initial state and a
+    state transition.
+
+    Parameters
+    ----------
+    initarg
+        Initial starting input of the process.
+    initrv
+        Random variable describing the initial state.
+    transition
+        State transition of the system.
+
+    See Also
+    --------
+    RandomProcess : Random processes.
+    GaussianProcess : Gaussian processes.
+    """
+
+    def __init__(
         self,
-        args: _InputType,
-        base_measure: Type[randvars.RandomVariable],
-        sample: np.ndarray,
-    ) -> np.ndarray:
-        raise NotImplementedError
+        *,
+        initarg: np.ndarray,
+        initrv: randvars.RandomVariable,
+        transition: continuous.SDE,
+    ):
+        if not isinstance(transition, continuous.SDE):
+            msg = "The transition is not continuous. Did you mean 'MarkovSequence'?"
+            raise TypeError(msg)
```

Comment:
I do not see the point of covering this exception with a test, so I would propose to ignore codecov here as well.

Reply:
I agree. See the comment above to resolve the codecov message :)